db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Retirement/StateMachines/Methods.class/__methods__/remove_from_provider.rb

```diff
@@ -13,7 +13,7 @@
   case removal_type
   when "remove_from_disk"
     $evm.log('info', "Removing VM:<#{vm.name}> from provider:<#{ems.try(:name)}>")
-    vm.remove_from_disk(false)
+    vm.remove_from_disk(false) if ems
```

Comment:
@tinaafitz you mean like this?
[app/models/vm_or_template.rb#L363](app/models/vm_or_template.rb#L363) is the safety net inside the vm.
It would raise a `RuntimeError` in case the ems is not present.

So, in case we check this edge case inside the StateMachine, doesnt that put too much logic into the StateMachine method?